test/acc_SUITE.erl

```diff
@@ -133,38 +126,92 @@ produce_iq_meta_automatically(_C) ->
-                              element => stanza_with_cdata() }),
+    Acc = mongoose_acc:new(?ACC_PARAMS#{element => stanza_with_cdata()}),
     {XMLNS, _} = mongoose_iq:xmlns(Acc),
     ?assertEqual(<<"jabber:iq:roster">>, XMLNS).
 
 strip(_C) ->
-    Acc = mongoose_acc:new(#{ location => ?LOCATION,
-                              lserver => <<"localhost">>,
-                              element => iq_stanza(),
-                              from_jid => jid:make(<<"jajid">>, <<"localhost">>, <<>>),
-                              to_jid => jid:make(<<"tyjid">>, <<"localhost">>, <<>>) }),
+    El = iq_stanza(),
+    FromJID = jid:make(<<"jajid">>, <<"localhost">>, <<>>),
+    ToJID = jid:make(<<"tyjid">>, <<"localhost">>, <<>>),
+    Server = maps:get(lserver, ?ACC_PARAMS),
+    HostType = maps:get(host_type, ?ACC_PARAMS),
+    Acc = mongoose_acc:new(?ACC_PARAMS#{element => El,
+                                        from_jid => FromJID,
+                                        to_jid => ToJID}),
     {XMLNS1, Acc1} = mongoose_iq:xmlns(Acc),
     ?assertEqual(<<"urn:ietf:params:xml:ns:xmpp-session">>, XMLNS1),
     ?assertEqual(<<"set">>, mongoose_acc:stanza_type(Acc1)),
     ?assertEqual(undefined, mongoose_acc:get(ns, ppp, undefined, Acc1)),
     Acc2 = mongoose_acc:set_permanent(ns, ppp, 997, Acc1),
-    Acc3 = mongoose_acc:set_permanent(ns2, [{a, 1}, {b, 2}], Acc2),
+    Acc3 = mongoose_acc:set(ns2, [{a, 1}, {b, 2}], Acc2),
     ?assertMatch([_, _], mongoose_acc:get(ns2, Acc3)),
-    Acc4 = mongoose_acc:delete(ns2, Acc3),
-    ?assertEqual(997, mongoose_acc:get(ns, ppp, Acc4)),
-    ?assertEqual([], mongoose_acc:get(ns2, Acc4)),
-    Ref = mongoose_acc:ref(Acc4),
-    NAcc1 = mongoose_acc:strip(#{ lserver => <<"localhost">>,
-                                 element => sample_stanza() }, Acc4),
+    ?assertEqual(Server, mongoose_acc:lserver(Acc3)),
+    ?assertEqual(HostType, mongoose_acc:host_type(Acc3)),
+    ?assertEqual({FromJID, ToJID, El}, mongoose_acc:packet(Acc3)),
+    Ref = mongoose_acc:ref(Acc3),
+    NewServer = <<"test.", Server/binary>>,
+    NewHostType = <<"new ",HostType/binary>>,
+    NAcc1 = mongoose_acc:strip(#{lserver => NewServer,
+                                 host_type => NewHostType,
+                                 element => sample_stanza() }, Acc3),
     {XMLNS2, NAcc2} = mongoose_iq:xmlns(NAcc1),
     ?assertEqual(<<"urn:xmpp:blocking">>, XMLNS2),
     ?assertEqual(jid:from_binary(<<"a@localhost">>), mongoose_acc:to_jid(NAcc2)),
     ?assertEqual(Ref, mongoose_acc:ref(NAcc2)),
     ?assertEqual(997, mongoose_acc:get(ns, ppp, NAcc2)),
-    ?assertEqual([], mongoose_acc:get(ns2, NAcc2)).
-
+    ?assertEqual([], mongoose_acc:get(ns2, NAcc2)),
+    ?assertEqual(sample_stanza(), mongoose_acc:element(NAcc2)),
+    ?assertEqual(NewServer, mongoose_acc:lserver(NAcc2)),
+    ?assertEqual(NewHostType, mongoose_acc:host_type(NAcc2)),
+    NAcc3 = mongoose_acc:strip(Acc3),
```

Comment:
At this point it becomes really hard to follow the test case. Please break it into smaller ones.